js/iframe-coordinator/src/messages/ClientToHost.ts

```diff
@@ -1,4 +1,20 @@
 import { LabeledPublication } from './Publication';
 import { LabeledToast } from './Toast';
+import { LabeledNavRequest } from './NavRequest';
 
-export type ClientToHost = LabeledPublication | LabeledToast;
+export type ClientToHost =
+  | LabeledPublication
+  | LabeledToast
+  | LabeledNavRequest;
+
+export function validate(msg: ClientToHost) {
+  //TODO: actually validate cases
+  switch (msg.msgType) {
+    case 'navRequest':
+      return msg;
+    case 'toastRequest':
+      return msg;
+    case 'publish':
+      return msg;
+  }
```

Comment:
Handling the incoming messages should also use a switch pattern like this, IMO, which will ensure at compile time that the host and client libraries have paths for every type of inbound message and that we haven't typo'd a label anywhere.